api.planx.uk/client/index.ts

```diff
@@ -11,3 +17,26 @@ export const $admin = new CoreDomainClient({
 export const $public = new CoreDomainClient({
   targetURL: process.env.HASURA_GRAPHQL_URL!,
 });
+
+/**
+ * Get a planx-core client with permissions scoped to the current user.
+ * This client instance ensures that all operations are restricted
+ * to the permissions of the user who initiated the request.
+ */
+export const getClient = () => {
```

Comment:
Not thrilled on the naming here - feels a little generic and meaningless.

Something like `getUserRoleClient()` might be more accurate but it's pretty wordy... 

No strong opinions here on naming, but if we move to a future where the API only calls a `$public` client (pretty clear what this is) and permissions-scoped `$client` that might actually be fine?